service-mango-orderbook/src/orderbook_filter.rs

```diff
@@ -202,85 +230,122 @@ pub async fn init(
+                            );
+                            last_write_versions.insert(
+                                oracle_pk_string.clone(),
+                                (oracle_info.slot, oracle_info.write_version),
+                            );
 
-                            let account = &account_info.account;
-                            let bookside =
-                                BookSide::try_deserialize(account.data().borrow_mut()).unwrap();
-                            let side = match bookside.nodes.order_tree_type() {
-                                OrderTreeType::Bids => OrderbookSide::Bid,
-                                OrderTreeType::Asks => OrderbookSide::Ask,
+                            let keyed_account = KeyedSharedDataAccountReader {
+                                key: oracle_pk,
+                                shared: oracle_info.account.clone(),
+                            };
+                            let oracle_config = OracleConfigParams {
+                                conf_filter: mkt.1.conf_filter,
+                                max_staleness_slots: mkt.1.max_staleness_slots,
                             };
-                            let time_now = SystemTime::now()
-                                .duration_since(UNIX_EPOCH)
-                                .unwrap()
-                                .as_secs();
-                            let oracle_price_lots = 0; // todo: does this matter? where to find it?
-                            let bookside = bookside
-                                .iter_valid(time_now, oracle_price_lots)
-                                .map(|item| (item.node.price_data() as i64, item.node.quantity))
-                                .group_by(|(price, _)| *price)
-                                .into_iter()
-                                .map(|(price, group)| {
-                                    [
-                                        price_lots_to_ui_perp(
-                                            price,
-                                            mkt.1.base_decimals,
-                                            mkt.1.quote_decimals,
-                                            mkt.1.base_lot_size,
-                                            mkt.1.quote_lot_size,
-                                        ),
-                                        base_lots_to_ui_perp(
-                                            group.map(|(_, quantity)| quantity).sum(),
-                                            mkt.1.base_decimals,
-                                            mkt.1.base_lot_size,
-                                        ),
-                                    ]
-                                })
-                                .collect();
 
-                            let other_bookside = bookside_cache.get(&other_side_pk.to_string());
+                            if let Ok((oracle_price, _slot)) = state::oracle_price_and_slot(
```

Comment:
Not sure about checking staleness here, think this will break all updates if the oracle goes stale not just oracle peg

Reply:
you're right, fixed